geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/SimpleFeatureValidator.scala

```diff
@@ -20,154 +21,155 @@ trait SimpleFeatureValidator {
+
+    override def init(sft: SimpleFeatureType): Unit = {
+      lastErr = null
+      dtgIndex = sft.getDtgIndex.getOrElse(-1)
+      geomIndex = Option(sft.getGeomField).map(sft.indexOf).getOrElse(-1)
+      minDate = Date.from(BinnedTime.ZMinDate.toInstant)
+      maxDate = Date.from(BinnedTime.maxDate(sft.getZ3Interval).toInstant)
 
-/**
-  * Validates that the geometry fits into a Z2 or Z3 style WGS84 index scheme
-  */
-class ZIndexValidator extends SimpleFeatureValidator {
-  private var valid: (SimpleFeature) => Boolean = _
-  private var lastErr: String = _
-
-  override def name: String = "z-index"
-
-  override def init(sft: SimpleFeatureType): Unit = {
-    valid = {
-      import org.locationtech.geomesa.utils.geotools.Conversions._
-      import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
-
-      val maxDate = Date.from(BinnedTime.maxDate(sft.getZ3Interval).toInstant)
-      val minDate = Date.from(BinnedTime.ZMinDate.toInstant)
-      val dtgFn: (SimpleFeature) => Boolean = sft.getDtgIndex match {
-        case Some(dtgIdx) => (sf: SimpleFeature) =>
-          lastErr = null
-          if (sf.getAttribute(dtgIdx) == null){
-            lastErr = "Null date attribute"
-            false
-          } else if (sf.get[Date](dtgIdx).before(minDate)) {
-            lastErr = "Date is before Z3 Min Date"
-            false
-          } else if (sf.get[Date](dtgIdx).after(maxDate)) {
-            lastErr = "Date is after Z3 Max Date"
-            false
-          } else {
-            true
-          }
-        case None =>
-          throw new IllegalArgumentException("Z3 validator cannot be used on a type lacking a dtg index")
+      if (dtgIndex == -1 || geomIndex == -1) {
+        throw new IllegalArgumentException("Z3 validator cannot be used on a type lacking a date or geometry")
       }
+    }
 
-      val geomFn: (SimpleFeature) => Boolean =
-        if (sft.getGeometryDescriptor != null) {
-          (sf: SimpleFeature) =>
-            lastErr = null
-            if (sf.getDefaultGeometry == null) {
-              lastErr = "Geom is null"
-              false
-            } else if (!org.locationtech.geomesa.utils.geotools.wholeWorldEnvelope.contains(sf.geometry.getEnvelopeInternal)) {
-              lastErr = "Geom is not contained in whole world envelope"
-              false
-            } else {
-              true
-            }
-        } else {
-          throw new IllegalArgumentException("Z3 validator cannot be used on a type lacking a geom index")
-        }
-
-      (sf: SimpleFeature) => dtgFn(sf) && geomFn(sf)
+    override def validate(sf: SimpleFeature): Boolean = {
+      lastErr = null
+      val date = sf.getAttribute(dtgIndex).asInstanceOf[Date]
+      val geom = sf.getAttribute(geomIndex).asInstanceOf[Geometry]
+      if (date == null) {
+        lastErr = "Null date attribute"
+        false
+      } else if(geom == null) {
```

Comment:
picky add space :)

Reply:
man, i had the right number of spaces, just in the wrong place ;)